src/serde/reader.rs

```diff
@@ -97,12 +97,8 @@ impl fmt::Display for DeserializeError {
 }
 
 impl serde::de::Error for DeserializeError {
-    fn custom<T: Into<String>>(desc: T) -> DeserializeError {
-        DeserializeError::Serde(serde::de::value::Error::Custom(desc.into()))
-    }
-
-    fn end_of_stream() -> DeserializeError {
-        DeserializeError::Serde(serde::de::value::Error::EndOfStream)
+    fn custom<T: ::std::fmt::Display>(desc: T) -> DeserializeError {
```

Comment:
The `fmt` module is already imported above so this could be `fmt::Display` instead of `::std::fmt::Display`.